simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/StationState.java

```diff
@@ -141,16 +142,26 @@ public boolean hasOngoingTransaction(Integer evseId) {
         return findEvse(evseId).hasOngoingTransaction();
     }
 
+    /**
+     * Try to find EVSE by given EVSE ID or return empty result.
+     *
+     * @param evseId EVSE identity
+     * @return optional with instance of {@link Evse} or Optional.empty()
+     */
+    public Optional<Evse> tryFindEvse(int evseId) {
+        return evses.stream()
+                .filter(evse -> evse.getId() == evseId)
+                .findAny();
+    }
+
     /**
      * Find an instance of {@link Evse} by evseId. If not found then throw {@link IllegalArgumentException}.
      *
-     * @param evseId evse identity
+     * @param evseId EVSE identity
      * @return an instance of {@link Evse}
      */
     public Evse findEvse(int evseId) {
```

Comment:
this method is only used inside this class right? no need for it to be public then.

Sorry, I know I said in the previous CR iteration that you shouldn't have a private method calling a public one, but if there isn't a better way to re-organize the code, then just leave it.

Reply:
no, it's used in many places